### PR TITLE
feat: add external logins redux actions

### PR DESCRIPTION
**Pull request by andreecosta**
## Description

This adds the missing actions and reducer for external logins endpoints.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/redux/src/users/authentication/actions/fetchUserExternalLogins.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { fetchUserExternalLoginsFactory } from './factories/index.js';
+import { getUserExternalLogins } from '@farfetch/blackout-client';
+
+/**
+ * Fetch the user external logins.
```
Comment (boliveira) [inline packages/redux/src/users/authentication/actions/fetchUserExternalLogins.ts:5]:
```suggestion
 * Fetches the user external logins.
```

File: packages/redux/src/users/authentication/__tests__/reducer.test.ts [hunk 2]
```diff
@@ -121,6 +123,129 @@ describe('authentication reducer', () => {
...
+    });
+
+    it('should handle other actions by returning the previous state', () => {
+      const state = {
+        externalLogins: {
+          result: null,
+          error: 'error',
+          isLoading: false,
+        },
+      } as unknown as AuthenticationState;
```
Comment (boliveira) [inline packages/redux/src/users/authentication/__tests__/reducer.test.ts:241]: Avoid having this please, you can easily have the state complying with AuthenticationState by using the fromReducer and having the `error` property be a `BlackoutError` instance

File: packages/redux/src/users/authentication/__tests__/selectors.test.ts [hunk 3]
```diff
@@ -134,4 +140,28 @@ describe('authentication redux selectors', () => {
       expect(selectors.isAuthenticated(mockState)).toBe(true);
     });
   });
+
+  describe('areExternalLoginsLoading()', () => {
```
Comment (boliveira) [inline packages/redux/src/users/authentication/__tests__/selectors.test.ts:144]: Missing test for `getExternalLogins` selector.

File: packages/redux/src/users/authentication/actions/fetchUserExternalLogins.ts (hunk 1 above)
Comment (andreecosta) [reply packages/redux/src/users/authentication/actions/fetchUserExternalLogins.ts]: Done!

File: packages/redux/src/users/authentication/__tests__/reducer.test.ts (hunk 2 above)
Comment (andreecosta) [reply packages/redux/src/users/authentication/__tests__/reducer.test.ts]: Sorry, missed it. Fixed.

File: packages/redux/src/users/authentication/__tests__/selectors.test.ts (hunk 3 above)
Comment (andreecosta) [reply packages/redux/src/users/authentication/__tests__/selectors.test.ts:144]: Doesn't the `describe('sub-areas with result', () => {}` tests for the result?
I've added the `ExternalLogins` sub-area to that test.

Comment (boliveira) [reply packages/redux/src/users/authentication/__tests__/selectors.test.ts:144]: You are right, my mistake.

**Review by boliveira — APPROVED** [review]

**Review by pedro-gbf — APPROVED** [review]

**Review by jpklzm — APPROVED** [review]

**Pull request merged**